tastebuddies/views.py

```diff
@@ -0,0 +1,372 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from collections import OrderedDict
+from cryptacular.bcrypt import BCRYPTPasswordManager
+import datetime
+
+from email.MIMEMultipart import MIMEMultipart
+from email.MIMEText import MIMEText
+
+from models import User, Group, Criteria
+import os
+
+from pyramid.httpexceptions import HTTPFound, HTTPForbidden
+from pyramid.security import remember, forget
+from pyramid.view import view_config
+
+from random import randint
+import smtplib
+
+
+@view_config(route_name='home',
+             renderer='templates/home.jinja2')
+def home_view(request):
+    return {'username': request.authenticated_userid}
+
+
+@view_config(route_name='user_create',
+             renderer='templates/user_create.jinja2')
+def user_create_view(request):
+
+    username = request.authenticated_userid
+
+    if request.method == 'POST':
+
+        try:
+            manager = BCRYPTPasswordManager()
+            username = request.params.get('username')
+            password = request.params.get('password')
+            hashed = manager.encode(password)
+            email = request.params.get('email')
+            user = User.add(
+                username=username,
+                password=hashed,
+                email=email,
+            )
+            Criteria.add(user=user)
+            headers = remember(request, username)
+
+            return HTTPFound(request.route_url('send_email'), headers=headers)
+        except:
+            return {}
+    return {'username': username}
+
+
+@view_config(route_name='send_email', permission='authn')
+def send_verify_email(request):
+
+    ver_code = randint(1000, 9999)
+
+    username = request.authenticated_userid
+    user = User.lookup_by_attribute(username=username)[0]
+    user.edit(id=user.id, ver_code=int(ver_code))
+
+    fromaddr = "tastebot@gmail.com"
+    toaddr = user.email
+
+    msg = MIMEMultipart()
+    msg["From"] = fromaddr
+    msg["To"] = toaddr
+    msg["Subject"] = "Your Tastebuddies Verification Code"
+
+    body = ''
+
+    here = os.path.dirname(os.path.abspath(__file__))
+    email_directory = os.path.join(here, 'static', 'email_templates')
+    body_template = os.path.join(email_directory, 'body.txt')
+    with open(body_template, 'r') as fh:
+        body = str(fh.read())
+
+    body = body.format(ver_code=ver_code)
+
+    msg.attach(MIMEText(body, 'plain'))
+
+    server = smtplib.SMTP('smtp.gmail.com', 587)
+    server.starttls()
+    server.login("tastebot", 'TASTEBUDDIES')
+    text = msg.as_string()
+
+    server.sendmail(fromaddr, toaddr, text)
```

Comment:
feels like everything in here that has to do with setting up a server and sending a message from it should probably be broken out into a separate module so that you can mock it during testing.  As this stands, it'd be hard to test this method without having to send an email somewhere (you need a working smtp endpoint to connect to).